src/main/java/com/github/nicholasmoser/gnt4/seq/groups/OpcodeGroup09.java

```diff
@@ -2,61 +2,130 @@
+      case 0x02 -> op_0902(bs);
+      case 0x03 -> dec(bs);
+      case 0x04 -> add(bs);
+      case 0x05 -> sub(bs);
+      case 0x06 -> subws(bs);
+      case 0x07 -> mov(bs);
+      case 0x08 -> o2p(bs);
+      case 0x09 -> p2o(bs);
+      case 0x0A -> push(bs);
+      case 0x0B -> pop(bs);
       case 0x0C -> op_090C(bs);
-      case 0x14 -> op_0914(bs);
-      case 0x1D -> op_091D(bs);
+      case 0x14 -> branch(bs);
+      case 0x15 -> branchEqualToZero(bs);
+      case 0x16 -> branchNotEqualToZero(bs);
+      case 0x17 -> branchGreaterThanZero(bs);
+      case 0x18, 0x1C -> branchGreaterThanEqualToZero(bs);
+      case 0x19, 0x1B -> branchLessThanZero(bs);
+      case 0x1A -> branchLessThanOrEqualToZero(bs);
+      case 0x1D -> branchLink(bs);
+      case 0x1E -> branchEqualZeroLink(bs);
+      case 0x1F -> branchNotEqualZeroLink(bs);
+      case 0x20 -> branchGreaterThanZeroLink(bs);
+      case 0x21, 0x25 -> branchGreaterThanEqualZeroLink(bs);
+      case 0x22, 0x24 -> branchLessThanZeroLink(bs);
+      case 0x23 -> branchLessThanEqualZeroLink(bs);
       case 0x26 -> op_0926(bs);
       default -> throw new IOException(String.format("Unimplemented: %02X", opcodeByte));
     };
   }
 
-  private static Opcode op_0901(ByteStream bs) throws IOException {
+  private static Opcode op_0900(ByteStream bs) throws IOException {
     int offset = bs.offset();
-    SEQ_RegCMD2 ea = SEQ_RegCMD2.get(bs);
+    SEQ_RegCMD1 ea = SEQ_RegCMD1.get(bs);
     String info = String.format(" %s", ea.getDescription());
     return new UnknownOpcode(offset, ea.getBytes(), info);
   }
 
-  private static Opcode op_0907(ByteStream bs) throws IOException {
+  private static Opcode movc(ByteStream bs) throws IOException {
     int offset = bs.offset();
     SEQ_RegCMD2 ea = SEQ_RegCMD2.get(bs);
     String info = String.format(" %s", ea.getDescription());
+    return new Movc(offset, ea.getBytes(), info);
+  }
+
+  private static Opcode op_0902(ByteStream bs) throws IOException {
+    int offset = bs.offset();
+    SEQ_RegCMD1 ea = SEQ_RegCMD1.get(bs);
+    String info = String.format(" %s", ea.getDescription());
     return new UnknownOpcode(offset, ea.getBytes(), info);
   }
 
-  private static Opcode op_0908(ByteStream bs) throws IOException {
+  private static Opcode dec(ByteStream bs) throws IOException {
     int offset = bs.offset();
     SEQ_RegCMD2 ea = SEQ_RegCMD2.get(bs);
     String info = String.format(" %s", ea.getDescription());
-    return new UnknownOpcode(offset, ea.getBytes(), info);
+    return new Decrement(offset, ea.getBytes(), info);
   }
 
-  private static Opcode op_0909(ByteStream bs) throws IOException {
+  private static Opcode add(ByteStream bs) throws IOException {
     int offset = bs.offset();
     SEQ_RegCMD2 ea = SEQ_RegCMD2.get(bs);
     String info = String.format(" %s", ea.getDescription());
-    return new UnknownOpcode(offset, ea.getBytes(), info);
+    return new Add(offset, ea.getBytes(), info);
```

Comment:
Ideally, I would prefer that every opcode has a unique mnemonic for it, so that it's clear which one is being used in the disassemble report. This opcode (and others in this class) currently overlaps with `0407`, also using the mnemonic `add`.

But on that note, it appears that you are correct. This opcode seems to be a duplicate of `0407`:

![image](https://user-images.githubusercontent.com/4983605/161663801-13678512-23b4-46d3-a73d-1840aaa022e3.png)

Thanks to Doraemon on the Wii having a symbol map, I actually know what the opcode group for group 09 and its siblings are.

| Group | Eighting Official Name | Use |
| --- | --- | --- |
| 04  | SEQ_CmdI | **I**nt operations (4 bytes) |
| 05  | SEQ_CmdIC | **C**har operations (1 byte) |
| 06  | SEQ_CmdIS | **S**hort operations (2 bytes) |
| 07  | SEQ_CmdIL | **L**ong operations (8 bytes) |
| 08  | SEQ_CmdF | **F**loat operations (4 bytes) |
| 09  | SEQ_CmdP | ??? |
| 0A  | SEQ_CmdIV | ??? |
| 0B  | SEQ_CmdFV | ??? |
| 0C  | SEQ_CmdFM | ??? |
| 0D  | SEQ_CmdRGB | Unused in GNT4 and Doraemon |

But I have no idea what the **P** stands for. Regardless, I think I would prefer to prepend all operations in `09` with whatever the P is for, like I've already done for opcodes like `ShortAdd.java` with the mnemonic `s_add`.

What do you think? My best guess currently is that the P stands for pointer.

Reply:
A cursory glance of SEQ files shows that the group `09` opcodes are more common than the group `04` opcodes. The only thing more common than an int is a pointer, which leads credence to that theory.

Reply:
I'm also realizing this is inconsistent with how I've already done this, but I think it makes more sense to do the operation followed by the value type. So `add_s` instead of `s_add`, so that it reads like **Add Short**

Reply:
This overall sounds good to me.

P for pointer also sounds very reasonable since it is used for a lot of branching to addresses stored in registers, and is used to store system memory pointers in both the seq_p, and the reg_p.

I could probably try and extend all the classes that would overlap, I'll get to clearing these out and then ping you with another review request.

With the naming I believe that the easiest to read order is
Operator type -> Operation name, as it is now
So for adding two direct values would be s_add/sadd/int16_add/i16_add/i16_t_add (sAdd feels kinda weird though) depending on verbosity and style choice, I'm kinda biased towards int16_add as the add and the type itself as int16_t, but that could very well be me getting used to networking libraries that try to stay platform agnostic.
The reason I prefer this order is that I think it is more important to quickly discern what type of operators are used than what operation is being done when trying to trace a path together with memory positions.

Reply:
int16_add and int16_t both sound good to me! Don't worry about changing this in all the other locations, I can open a pr later with this syntax for opcode groups other than 09